src/client/websocket/WebSocketShard.js

```diff
@@ -475,12 +475,20 @@ class WebSocketShard extends EventEmitter {
       return;
     }
     const hasGuildsIntent = new Intents(this.manager.client.options.intents).has(Intents.FLAGS.GUILDS);
-    // Step 2. Create a 15s timeout that will mark the shard as ready if there are still unavailable guilds
+    // Step 2. Create a timeout that will mark the shard as ready if there are still unavailable guilds
+    // * The timeout is 15 seconds by default
+    // * This can be optionally be changed in the client options via the `fetchGuildTimeout` option
+    // * a timeout time of zero will skip this timeout, which potentially could cause the Client to miss guilds.
+
+    const guildTimeoutAmount = this.manager.client.options.fetchGuildTimeout ?? 15000;
```

Comment:
```suggestion
    const guildTimeoutAmount = this.manager.client.options.fetchGuildTimeout;
```

Reply:
Defaults will be assigned in the client constructor when not specified, so there's no need to specify it here as well